sky/cli.py

```diff
@@ -342,6 +343,142 @@ def _parse_env_var(env_var: str) -> Tuple[str, str]:
 ]
 
 
+def _complete_cluster_name(ctx: click.Context, param: click.Parameter,
+                           incomplete: str) -> List[str]:
+    """Handle shell completion for cluster names."""
+    del ctx, param  # Unused.
+    return global_user_state.get_cluster_names(incomplete)
+
+
+def _complete_storage_name(ctx: click.Context, param: click.Parameter,
+                           incomplete: str) -> List[str]:
+    """Handle shell completion for storage names."""
+    del ctx, param  # Unused.
+    return global_user_state.get_storage_names(incomplete)
```

Comment:
nit: for readability, it would be better to rename this function to global_user_state.get_storage_names_start_with() so as to avoid ambiguity.